ast/src/parsed/mod.rs

```diff
@@ -239,6 +249,57 @@ impl Children<Expression> for PilStatement {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct StructDeclaration<E = u64> {
+    pub name: String,
+    pub type_vars: TypeBounds,
+    pub fields: Vec<NamedType<E>>,
+}
+
+impl<E> StructDeclaration<E> {
+    pub fn type_of_field(&self, name: &str) -> Option<&Type<E>> {
```

Comment:
Not sure how this is used, but it could actually return a TypeScheme that contains the type vars of the struct